src/contrib/mlir/core/ngraph_dialect/ops.cpp

```diff
@@ -172,7 +172,8 @@ static mlir::LogicalResult verifyCmpOp(T op)
 
     // result of same shape as input and has bool type
     if (!resType.isCompatibleShape(opType0) ||
-        !resType.getElementType().cast<NGIntegerType>().isUInt8())
+        !(resType.getElementType().cast<NGIntegerType>().isUnsigned() &&
+          resType.getElementType().cast<NGIntegerType>().getWidth() == 8))
```

Comment:
Perhaps better to have NGIntegerType as a wrapper around mlir Integer type. To keep helpers like isUInt8() and other checks instead of using literals.

Reply:
Fixed by using combined signedness/type is_type getters.  See latest commit.  Resolving.